NuRadioReco/modules/io/RNO_G/readRNOGDataMattak.py

```diff
@@ -0,0 +1,642 @@
+         
+      mattak_backend: str
+         Select a mattak backend. Options are "auto", "pyroot", "uproot". If "auto" is selected, pyroot is used if available otherwise
+         a "fallback" to uproot is used. (Default: "auto") 
+
+      log_level: enum
+         Set verbosity level of logger
+      """
+      
+      t0 = time.time()
+      
+      self.logger = logging.getLogger('NuRadioReco.readRNOGData')
+      self.logger.setLevel(log_level)
+      
+      self._read_calibrated_data = read_calibrated_data
+      self._apply_baseline_correction = apply_baseline_correction
+      self._convert_to_voltage = convert_to_voltage
+      
+      # Temporary solution hard-coded values from Cosmin. Only used when uncalibrated data
+      # is read and convert_to_voltage is True.
+      self._adc_ref_voltage_range = 2.5 * units.volt
+      self._adc_n_bits = 12
+      
+      self.__max_trigger_rate = max_trigger_rate
+      self.__run_types = run_types
+      
+      if run_time_range is not None:
+         convert_time = lambda t: None if t is None else astropy.time.Time(t)
+         self._time_low = convert_time(run_time_range[0])
+         self._time_high = convert_time(run_time_range[1])
+      else:
+         self._time_low = None
+         self._time_high = None         
+      
+      if select_runs:
+         if run_table_path is None:
+            global imported_runtable
+            if imported_runtable:
+               self.logger.debug("Access RunTable database ...")
+               try:
+                  self.__run_table = RunTable().get_table()
+               except:
+                  self.logger.error("No connect to RunTable database could be established. "
+                                    "Runs will not be filtered.")
+                  imported_runtable = False
+         else:
+            self.__run_table = pandas.read_csv(run_table_path)
+            imported_runtable = True
+            
+      if select_runs:
+         self.logger.info("\n\tSelect runs with type: {}".format(", ".join(run_types)) +
+                         f"\n\tSelect runs with max. trigger rate of {max_trigger_rate / units.Hz} Hz"
+                         f"\n\tSelect runs which are between {self._time_low} - {self._time_high}")
+      
+      if not isinstance(data_dirs, (list, np.ndarray)):
+         data_dirs = [data_dirs]
+
+      if selectors is not None:
+         if not isinstance(selectors, (list, np.ndarray)):
+            selectors = [selectors]
+         
+         self.logger.info(f"Found {len(selectors)} selector(s)")
+
+      self._selectors = selectors
+            
+      if select_triggers is not None:
+         if isinstance(select_triggers, str):
+            selectors.append(lambda eventInfo: eventInfo.triggerType == select_triggers)
+         else:
+            for select_trigger in select_triggers:
+               selectors.append(lambda eventInfo: eventInfo.triggerType == select_trigger)
```

Comment:
I'm noticing more stuff now that I'm implementing this for the data monitor... I think this order should be swapped, because it's `self._selectors` and not `selectors` that's used in the other methods, right?

I also wonder if it's worth implementing a `set_selectors` method (or changing where/how the selectors are implemented). E.g. in my use case I know in advance which runs I want to open, but I might then want to change only the selectors to look at different events. It feels slightly unintuitive to call `begin` every time in that case (+, on a mounted file system, the `os.path.exists` calls are not free either).

Reply:
Sure seems useful.

Reply:
I resolved this comment. I fixed the bug. The set_selectors interface can be included in the next PR (which I am already working on)